qa/tasks/cephfs/test_scrub_checks.py

```diff
@@ -362,6 +364,74 @@ def test_scrub_repair(self):
         # fragstat should be fixed
         self.mount_a.run_shell(["rmdir", test_dir])
 
+    def test_scrub_merge_dirfrags(self):
+        """
+        That a directory is merged during scrub.
+        """
+
+        abs_test_path = "testdir"
+        split_size = 20
+        merge_size = 5
+        split_bits = 1
+        self.config_set('mds', 'mds_bal_split_size', split_size)
+        self.config_set('mds', 'mds_bal_merge_size', merge_size)
+        self.config_set('mds', 'mds_bal_split_bits', split_bits)
+
+        self.mount_a.run_shell(["mkdir", abs_test_path])
+        dir_ino=self.mount_a.path_to_ino(abs_test_path)
+
+        self.assertEqual(len(self.get_dsplits(dir_ino)), 0)
+        self.mount_a.create_n_files("{0}/file".format(abs_test_path), split_size * 2)
+
+        self.mount_a.umount_wait()
+
+        self.fs.flush()
+        self.fs.restart_mds()
+        self.fs.wait_for_daemons()
+
+        split_size = 100
+        merge_size = 30
+        self.config_set('mds', 'mds_bal_split_size', split_size)
+        self.config_set('mds', 'mds_bal_merge_size', merge_size)
+
+        self.assertEqual(len(self.get_dsplits(dir_ino)), 1)
+        self.fs.run_scrub(["start", "/{0}".format(abs_test_path), "recursive"])
```

Comment:
ditto

Reply:
same in `test_scrub_split_dirfrags()`

Reply:
Done. As you can see, I added `test_path` and then reworked `abs_test_path` to include the leading slash.